fastestimator/pipeline.py

```diff
@@ -112,43 +114,47 @@ def benchmark(self, mode: str = "train", epoch: int = 0, num_steps: int = 1000,
             loader = loader.take(num_steps)
         start = time.perf_counter()
         for idx, _ in enumerate(loader):
-            if idx == num_steps:
-                break
-            if idx % log_interval == 0 and idx > 0:
+            if idx + 1 % log_interval == 0:
```

Comment:
enumerate takes a second arg saying where to start the idx. enumerate(loader, 1) rather than adjusting the idx after the fact

Reply:
fixed